WordPress/Sniffs/CodeAnalysis/AssignmentInConditionSniff.php

```diff
@@ -0,0 +1,184 @@
+		unset( $this->assignment_tokens[ T_DOUBLE_ARROW ] );
+
+		$starters                       = Tokens::$booleanOperators;
+		$starters[ T_SEMICOLON ]        = T_SEMICOLON;
+		$starters[ T_OPEN_PARENTHESIS ] = T_OPEN_PARENTHESIS;
+
+		$this->condition_start_tokens = $starters;
+
+		return array(
+			T_IF,
+			T_ELSEIF,
+			T_FOR,
+			T_SWITCH,
+			T_CASE,
+			T_WHILE,
+		);
+
+	}//end register()
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 *
+	 * @return void
+	 */
+	public function process_token( $stackPtr ) {
+
+		$token = $this->tokens[ $stackPtr ];
+
+		// Find the condition opener/closer.
+		if ( T_FOR === $token['code'] ) {
+			if ( isset( $token['parenthesis_opener'], $token['parenthesis_closer'] ) === false ) {
+				return;
+			}
+
+			$semicolon = $this->phpcsFile->findNext( T_SEMICOLON, ( $token['parenthesis_opener'] + 1 ), $token['parenthesis_closer'] );
+			if ( false === $semicolon ) {
+				return;
+			}
+
+			$opener    = $semicolon;
+			$semicolon = $this->phpcsFile->findNext( T_SEMICOLON, ( $opener + 1 ), $token['parenthesis_closer'] );
+			if ( false === $semicolon ) {
+				return;
+			}
+
+			$closer = $semicolon;
+			unset( $semicolon );
+
+		} elseif ( T_CASE === $token['code'] ) {
+			if ( isset( $token['scope_opener'] ) === false ) {
+				return;
+			}
+
+			$opener = $stackPtr;
+			$closer = $token['scope_opener'];
+
+		} else {
+			if ( isset( $token['parenthesis_opener'], $token['parenthesis_closer'] ) === false ) {
+				return;
+			}
+
+			$opener = $token['parenthesis_opener'];
+			$closer = $token['parenthesis_closer'];
+		}
+
+		$startPos = $opener;
+
+		do {
+			$hasAssignment = $this->phpcsFile->findNext( $this->assignment_tokens, ( $startPos + 1 ), $closer );
+			if ( false === $hasAssignment ) {
+				return;
+			}
+
+			// Examine whether the left side is a variable.
+			$hasVariable       = false;
+			$conditionStart    = $startPos;
+			$altConditionStart = $this->phpcsFile->findPrevious(
+				$this->condition_start_tokens,
+				( $hasAssignment - 1 ),
+				$startPos
+			);
+			if ( false !== $altConditionStart ) {
+				$conditionStart = $altConditionStart;
+			}
+
+			for ( $i = $hasAssignment; $i > $conditionStart; $i-- ) {
+				if ( isset( Tokens::$emptyTokens[ $this->tokens[ $i ]['code'] ] ) ) {
+					continue;
+				}
+
+				// If this is a variable or array, we've seen all we need to see.
+				if ( T_VARIABLE === $this->tokens[ $i ]['code']
+					|| T_CLOSE_SQUARE_BRACKET === $this->tokens[ $i ]['code']
+				) {
+					$hasVariable = true;
+					break;
+				}
+
+				// If this is a function call or something, we are OK.
+				if ( T_CLOSE_PARENTHESIS === $this->tokens[ $i ]['code'] ) {
+					break;
+				}
+			}
+
+			if ( true === $hasVariable ) {
+				$this->phpcsFile->addWarning(
+					'Variable assignment found within a condition. Did you mean to do a comparison ?',
```

Comment:
I'll change this, but be aware that the upstream version has been merged already and does include the space, so the message will change back by the time we switch over to the upstream version (unless someone changes it there too).

To be honest, most of the time it takes some time before something is merged upstream, so I thought I'd still have time to update the PR there after pulling it here, but for some reason this one got merged *very* quickly upstream ;-)